application/classes/Ushahidi/Repository/Post/Export.php

```diff
@@ -8,90 +8,89 @@
+			->select([DB::expr('DISTINCT(posts.form_id)'), 'form_id'])->execute($this->db);
+		$result =  $result->as_array();
+		return array_column($result, 'form_id');
+	}
+
 	/**
 	 * @param $data
 	 * @return array
 	 */
-  public function retrieveColumnNameData($data) {
-
-    /**
-     * Tags (native) should not be shown in the CSV Export
-    */
-	  unset($data['tags']);
-    // Set attribute keys
-    $attributes = [];
-  	foreach ($data['values'] as $key => $val)
-    {
-        $attribute = $this->form_attribute_repo->getByKey($key);
-        $attributes[$key] = ['label' => $attribute->label, 'input' => $attribute->input, 'priority'=> $attribute->priority, 'stage' => $attribute->form_stage_id, 'type'=> $attribute->type, 'form_id'=> $data['form_id']];
-
-        // Set attribute names. This is for categories (custom field) to show their label and not the ids
-        if ($attribute->type === 'tags') {
-          $data['values'][$key] = $this->retrieveTagNames($val);
-        }
-    }
-
-    $data += ['attributes' => $attributes];
-
-
-    // Set Set names
-    if (!empty($data['sets'])) {
-        $data['sets'] = $this->retrieveSetNames($data['sets']);
-    }
+	public function retrieveMetaData($data, $attributes)
+	{
 
-    // Get contact
-    if (!empty($data['contact_id'])) {
-        $contact = $this->contact_repo->get($data['contact_id']);
-        $data['contact_type'] = $contact->type;
-        $data['contact'] = $contact->contact;
-    }
+		/**
+		 * Tags (native) should not be shown in the CSV Export
+		 */
+		unset($data['tags']);
+		
+		// Set tag labels
+		foreach ($data['values'] as $key => $val) {
+			// Set attribute names. This is for categories (custom field) to show their label and not the ids
+			if (isset($attributes[$key]) && $attributes[$key]['type'] === 'tags') {
+				$data['values'][$key] = $this->retrieveTagNames($val);
+			}
+		}
 
-    // Set Completed Stage names
-    if(!empty($data['completed_stages'])) {
-      $data['completed_stages'] = $this->retrieveCompletedStageNames($data['completed_stages']);
-    }
+		// Get contact
+		if (!empty($data['contact_id'])) {
+			$contact = $this->contact_repo->get($data['contact_id']);
+			$data['contact_type'] = $contact->type;
+			$data['contact'] = $contact->contact;
+		}
 
-    // Set Form name
-    if (!empty($data['form_id'])) {
-      $form = $this->form_repo->get($data['form_id']);
-      $data['form_name'] = $form->name;
-    }
-    return $data;
-  }
+		// Set Form name
+		if (!empty($data['form_id'])) {
+			$form = $this->form_repo->get($data['form_id']);
+			$data['form_name'] = $form->name;
+		}
+		return $data;
+	}
 
-  public function retrieveTagNames($tag_ids) {
-    $tag_repo = service('repository.tag');
-    $names = [];
-    foreach($tag_ids as $tag_id) {
-      $tag = $tag_repo->get($tag_id);
-      array_push($names, $tag->tag);
-    }
-    return $names;
-  }
+	public function retrieveTagNames($tag_ids)
+	{
+		$tag_repo = service('repository.tag');
```

Comment:
You should inject this. Using `service` should be an exception and mostly only appear in controllers/commands, not in the core repos